### PR TITLE
Fix initialization of storageClass lister

### LINKED ISSUE #395: Nil-pointer dereference when deleting volume
This happens on HEAD/canary. If you use a CSI driver that doesn't support topology, and create a storage class with CSI secrets, and create a volume, then later try to delete that volume, the sidecar panics.

<pre>I0111 22:01:44.396387       1 controller.go:1311] delete "pvc-67d84e2f-397f-42ba-b013-e404a7f94a62": started
E0111 22:01:44.400326       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 159 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1501b60, 0x2456e30)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1501b60, 0x2456e30)
	/home/travis/.gimme/versions/go1.13.3.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/kubernetes-csi/external-provisioner/pkg/controller.(*csiProvisioner).Delete(0xc00002c3c0, 0xc0000f8f00, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/pkg/controller/controller.go:918 +0x2e1
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).deleteVolumeOperation(0xc000190900, 0xc0000f8f00, 0x2475920, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:1326 +0x25d
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).syncVolume(0xc000190900, 0x16ce5e0, 0xc0000f8f00, 0x16ce5e0, 0xc0000f8f00)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:1045 +0xaa
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).syncVolumeHandler(0xc000190900, 0xc0002b5440, 0x28, 0xc000373a78, 0x10)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:992 +0x96
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).processNextVolumeWorkItem.func1(0xc000190900, 0x148f860, 0xc0004c5170, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:934 +0xf9
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).processNextVolumeWorkItem(0xc000190900, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:951 +0x53
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).runVolumeWorker(0xc000190900)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:864 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0004b6710)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004b6710, 0x3b9aca00, 0x0, 0xc0001a7001, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004b6710, 0x3b9aca00, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).Run.func1
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:816 +0x3b9
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1374731]

goroutine 159 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1501b60, 0x2456e30)
	/home/travis/.gimme/versions/go1.13.3.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/kubernetes-csi/external-provisioner/pkg/controller.(*csiProvisioner).Delete(0xc00002c3c0, 0xc0000f8f00, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/pkg/controller/controller.go:918 +0x2e1
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).deleteVolumeOperation(0xc000190900, 0xc0000f8f00, 0x2475920, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:1326 +0x25d
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).syncVolume(0xc000190900, 0x16ce5e0, 0xc0000f8f00, 0x16ce5e0, 0xc0000f8f00)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:1045 +0xaa
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).syncVolumeHandler(0xc000190900, 0xc0002b5440, 0x28, 0xc000373a78, 0x10)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:992 +0x96
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).processNextVolumeWorkItem.func1(0xc000190900, 0x148f860, 0xc0004c5170, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:934 +0xf9
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).processNextVolumeWorkItem(0xc000190900, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:951 +0x53
sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).runVolumeWorker(0xc000190900)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:864 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0004b6710)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004b6710, 0x3b9aca00, 0x0, 0xc0001a7001, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004b6710, 0x3b9aca00, 0x0)
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/sig-storage-lib-external-provisioner/controller.(*ProvisionController).Run.func1
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:816 +0x3b9</pre>

Upon investigation, it appears that the csiProvisioner.scLister field was nil, because it was never set. Thus the attempt to get the storage class causes a dereference of this nil pointer, which is bad. 

For reasons I don't understand, the storage class lister is only initialized when the plugin supports topology. See cmd/csi-provisioner/csi-provisioner.go:203.

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

This PR fixes a nil pointer dereference due to the storageClass lister being initialized in the wrong place (i.e., only when topology was supported). This bug was introduced in https://github.com/kubernetes-csi/external-provisioner/commit/0c73acbb0b576c42319dc06692da61cd68ba9b75.

I tested this fix with csi-hostpath driver.

**Which issue(s) this PR fixes**:

Fixes #395

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/csi-provisioner/csi-provisioner.go [hunk 1]
```diff
@@ -193,14 +192,13 @@ func main() {
 		provisionerOptions = append(provisionerOptions, controller.AdditionalProvisionerNames([]string{supportsMigrationFromInTreePluginName}))
 	}
 
-	var scLister storagelistersv1.StorageClassLister
+	// Create informer to prevent hit the API server for all resource request
+	factory := informers.NewSharedInformerFactory(clientset, ctrl.ResyncPeriodOfCsiNodeInformer)
+	scLister := factory.Storage().V1().StorageClasses().Lister()
```
Comment (msau42) [inline cmd/csi-provisioner/csi-provisioner.go]: Do we have a unit test that could cover this?

Comment (bertinatto) [reply cmd/csi-provisioner/csi-provisioner.go]: Apparently not, this whole `main()` function isn't unit-tested.

File: cmd/csi-provisioner/csi-provisioner.go [hunk 2]
```diff
@@ -234,14 +232,12 @@ func main() {
...
-			for _, v := range cacheSyncResult {
-				if !v {
-					klog.Fatalf("Failed to sync Informers!")
-				}
+		stopCh := context.Background().Done()
+		factory.Start(stopCh)
+		cacheSyncResult := factory.WaitForCacheSync(stopCh)
+		for _, v := range cacheSyncResult {
+			if !v {
+				klog.Fatalf("Failed to sync Informers!")
```
Comment (bertinatto) [inline cmd/csi-provisioner/csi-provisioner.go]: I noticed that `factory` can no longer be `nil`, so I remove this check.

**Pull request merged**